src/main.cpp

```diff
@@ -2,36 +2,56 @@
 
 #include "acceleration.hpp"
 #include "barometer.hpp"
+#include "devices.hpp"
 #include "gps.hpp"
+#include "logger.hpp"
 
 constexpr auto baud_rate = 9600l;
 constexpr int interval = 2000;
-HardwareSerial Serial3{PC11, PC10};
+bool date_append = false;
+
+String flight_path_folder;
 
+HardwareSerial Serial3{PC11, PC10};
 MMA8452Q accelerometer;
 BMP280 bmp;
 Adafruit_GPS gps{&Serial3};
 
 uint32_t timer = millis();
 
+File file;
+
 void setup()
 {
     Serial.begin(baud_rate);
     Serial.println("setup");
-    obc::init(accelerometer).expect("accelerometer init failure");
-    obc::init(bmp).expect("BMP init failure");
-    obc::init(gps).expect("GPS init failure");
+
+    obc::init();
 }
 
 void loop()
 {
     if (timer - millis() > interval) {
+        obc::Packet logs = {{}, {}, {}, {}, {}};
         const auto acclr = obc::measure(accelerometer);
         const auto bmp_measurements = obc::measure(bmp);
-        const auto gps_measurments = obc::measure(gps);
-        if (acclr.is_ok()) { obc::print(acclr.unwrap()); }
-        if (bmp_measurements.is_ok()) { obc::print(bmp_measurements.unwrap()); }
-        if (gps_measurments.is_ok()) { obc::print(gps_measurments.unwrap()); }
+        logs.time = obc::read_time(gps);
+        logs.position = obc::read_position(gps);
+
+        if (acclr.is_ok()) { logs.acclr_measurements = acclr.unwrap(); }
+        if (bmp_measurements.is_ok()) {
+            logs.bmp_measurements = bmp_measurements.unwrap();
+        }
+
+        if (!date_append) {
+            date_append = true;
+            obc::file_appendln(
+                "/logs.csv",
+                obc::serialize(obc::read_date(gps)));
+        }
```

Comment:
```
if (csv_append_date) {
    csv_append_date = false;
    ...
```
a bardziej elegancko to przenieść to wszystko do setup(), który wykonuje się tylko raz - i tam zrobić check czy dane z gps są valid i fix złapany żeby nie dodać samych zer.